backend/wmg/data/load_corpus.py

```diff
@@ -177,43 +182,93 @@ def get_X_raw(anndata_object: anndata.AnnData) -> Union[np.ndarray, sparse.spmat
+
+            rankit_integrated_coo_matrix = filter_out_rankits_with_low_expression_counts(
+                rankit_integrated_csr_matrix, raw_expression_csr_matrix, expect_majority_filtered=True
             )
-            rankit_data = rankit_normalized_coo_sparse_raw_expression_matrix.data
 
-            array[rows, cols] = {"data": raw_data, "rankit": rankit_data}
+            global_rows = rankit_integrated_coo_matrix.row + start + first_obs_idx
+            global_cols = global_var_index[rankit_integrated_coo_matrix.col]
+
+            rankit_data = rankit_integrated_coo_matrix.data
+
+            assert len(rankit_data) == len(global_rows)
+            assert len(rankit_data) == len(global_cols)
+
+            array[global_rows, global_cols] = {"rankit": rankit_data}
             del (
-                coo_sparse_raw_expression_matrix,
-                rankit_normalized_coo_sparse_raw_expression_matrix,
-                rows,
-                cols,
-                raw_data,
+                raw_expression_csr_matrix,
+                rankit_integrated_coo_matrix,
+                rankit_integrated_csr_matrix,
+                global_rows,
+                global_cols,
                 rankit_data,
             )
             gc.collect()
 
     logger.debug(f"Saved {array_name}.")
+
+
+def filter_out_rankits_with_low_expression_counts(
+    rankits: csr_matrix, raw_counts_csr: csr_matrix, expect_majority_filtered=True
+) -> coo_matrix:
+    """
+    Keep only rankit values that were computed from expression values above the desired threshold.
+
+    @param expect_majority_filtered: Set this to True if the caller expects the majority of rankit values will be
+    filtered, as we can then use an optimal implementation
+    """
+
+    rankits_nnz_orig = rankits.nnz
+    raw_counts = raw_counts_csr.tocoo(copy=False)
+    to_keep_mask = raw_counts.data > RANKIT_RAW_EXPR_COUNT_FILTERING_MIN_THRESHOLD
+
+    # Unfortunately, it seems to take longer to decide which strategy to use than it does to just run either one.
+    # start = time.time()
+    # expect_majority_filtered = to_keep_mask.sum() < rankits_nnz_orig / 2
+    # end = time.time()
+    # logger.info(f"filter_out_rankits_with_low_expression_counts(): use extract strategy={expect_majority_filtered}; "
+    #             f"decision time={end - start}")
+
+    start = time.time()
+
+    if expect_majority_filtered:
+        rows_to_keep = raw_counts.row[to_keep_mask]
+        cols_to_keep = raw_counts.col[to_keep_mask]
+        rankits_to_keep = rankits.data[to_keep_mask]
+        rankits_filtered = coo_matrix((rankits_to_keep, (rows_to_keep, cols_to_keep)))
+    else:
+        to_filter_mask = ~to_keep_mask
+        rows_to_filter = raw_counts.row[to_filter_mask]
+        cols_to_filter = raw_counts.col[to_filter_mask]
+        rankits[rows_to_filter, cols_to_filter] = 0.0
+        rankits.eliminate_zeros()
+        rankits_filtered = rankits.tocoo()
```

Comment:
this `else` block is currently unused, but I couldn't bring myself to delete it just yet. 😄